instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_trace.rb

```diff
@@ -73,26 +126,18 @@ def execute_query_lazy(query:, multiplex:, &block)
 
           def execute_field(field:, query:, ast_node:, arguments:, object:, &block)
             platform_key = _otel_execute_field_key(field: field)
-            return super unless platform_key
```

Comment:
I can't explain why, but replacing this `super` call (and the one below in `execute_field_lazy`) made up a big part of the memory improvement. In the "before" profile, these lines appear in the result: 

```
       345  /Users/rmosolgo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/graphql-2.1.6/lib/graphql/schema/object.rb:82
       345  /Users/rmosolgo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/opentelemetry-instrumentation-graphql-0.26.7/lib/opentelemetry/instrumentation/graphql/tracers/graphql_trace.rb:104
       344  /Users/rmosolgo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/graphql-2.1.6/lib/graphql/execution/interpreter/runtime.rb:600
       344  /Users/rmosolgo/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/opentelemetry-instrumentation-graphql-0.26.7/lib/opentelemetry/instrumentation/graphql/tracers/graphql_trace.rb:76
       343  otel_bench.rb:104
```

But adding these arguments caused them to _not_ appear anymore. 

Looking at the breakdown by class, my only guess is that, somewhere under the hood, Ruby was making a `Hash` as part of this code: 

```diff 
  allocated objects by class
  -----------------------------------
-     6230  Hash
+     5126  Hash
      2395  Array
-     1188  Proc
+     1194  Proc
-      871  String
+      875  String
       785  Thread::Mutex
       784  OpenTelemetry::Context
       784  OpenTelemetry::SDK::Trace::Samplers::Result
       784  OpenTelemetry::SDK::Trace::Span
       784  OpenTelemetry::Trace::SpanContext
       423  GraphQL::Execution::Lazy
       345  GraphQL::Execution::Interpreter::Runtime::GraphQLResultHash
       301  MySchema::Book
```

Reply:
Thanks for sharing this. I wonder if there is a performance Rubocop linter we can leverage here.

Reply:
There is the [`rubocop-performance` gem](https://github.com/rubocop/rubocop-performance/), but after doing a quick test with the current instrumentation, these issues weren't pointed out.